File: remove_zero_sum_consecutive_nodes_from_linked_list.cpp
class Solution {
public:
    ListNode* removeZeroSumSublists(ListNode* head) {
        ListNode* dummyHead = new ListNode(0);
        dummyHead->next = head;
        ListNode* curr = dummyHead;

        while(curr != nullptr) {
            ListNode* ptr = curr->next;
            int sum = 0;

            while(ptr != nullptr) {
                sum += ptr->val;
                if(sum == 0) {
                    curr->next = ptr->next;
                }
                ptr = ptr->next;
            }
            curr = curr->next;
        }
        return dummyHead->next;
    }
};

// Time Complexity : O(n^2)
// Space Complexity : O(1)

File: remove_zero_sum_consecutive_nodes_from_linked_list.go
func removeZeroSumSublists(head *ListNode) *ListNode {
    dummyHead := &ListNode{Val: 0}
    dummyHead.Next = head
    curr := dummyHead

    for curr != nil {
        ptr := curr.Next
        sum := 0

        for ptr != nil {
            sum += ptr.Val
            if sum == 0 {
                curr.Next = ptr.Next
            }
            ptr = ptr.Next
        }
        curr = curr.Next
    }
    return dummyHead.Next
}

// Time Complexity : O(n^2)
// Space Complexity : O(1)